index.d.ts

```diff
@@ -331,3 +331,21 @@ export const applyDefaultProps: <P extends object>(
   oldProps: P,
   newProps: P
 ) => void;
+
+/**
+ * Register Component
+ *
+ * @param {string} type
+ * @param {Object} lifecycle methods
+ */
+export function registerComponent(type: string, lifecycle: _ReactPixi.ICustomComponent<{ [key: string]: any }, PIXI.DisplayObject>): void;
+
+/**
+ * Override existing components.
+ *
+ * This is not recommended usage, please ensure you know what you are doing before using this!
+ *
+ * @param {string} type
+ * @param {Object} lifecycle methods
+ */
+export function overrideComponent(type: string, lifecycle: _ReactPixi.ICustomComponent<{ [key: string]: any }, PIXI.DisplayObject>): void;
```

Comment:
Please remove this method, overriding components should not be allowed.

Could you add a `unregisterComponent` as well.. that does basically:

```js
export function unregisterComponent(type) { 
  delete TYPES_INJECTED[type];
}
```